pkgs/servers/overleaf/default.nix

```diff
@@ -0,0 +1,145 @@
+{ pkgs
+, lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, python3
+, fetchYarnDeps
+, fixup_yarn_lock
+, nodejs-16_x
+, yarn
+}:
+
+stdenv.mkDerivation rec {
+  pname = "overleaf";
+  version = "3.5.0";
+
+  src = fetchFromGitHub {
+    owner = "overleaf";
+    repo = "overleaf";
+    rev = "d539aaf2262149361590514fe27fd5c8e0fab111";
+    hash = "sha256-HuK9s7040sH9lBHb6yGO+fll2hrpdTIGIYIUujVkhqw=";
+  };
+
+  patches = [ ./versions.patch ];
+
+  postPatch = ''
+    # Add the custom yarn.lock
+    rm package-lock.json
+    cp ${./yarn.lock} yarn.lock
+    chmod +rw yarn.lock
+    cp ${./package.json} package.json
+
+    # Replace hard-coded values in settings by environment variables
+    find . -type f -not -path '*/\.*' -exec sed -E -i "s|SHARELATEX_|OVERLEAF_|g" {} +
+    (
+      cd server-ce/config
+      sed -E -i "s!'dockerhost',!undefined,\n      path: process.env.OVERLEAF_REDIS_PATH || undefined,!" ./settings.js
```

Comment:
How feasible do you think we can send a upstream patch for those?

Reply:
Probably not feasible, we are kinda tweaking the files they use to create their docker image: we replace the ``SHARELATEX_*`` env variables by ``OVERLEAF_*``, this is just cosmetic not to confuse the module users.
We also tweak the ``settings.js`` for more customizations. The overleaf ``settings.js`` file assumes running into docker and doesn't expose all options (for example when not running in a docker, it is nice to be able to change to web interface port, the port 3000 could be used by something else). The other alternative to using this script would be to commit our own ``settings.js`` to nixpkgs, which would be cleaner in my opinion but would have to merge conflicts at each update.

Reply:
I've submitted a patch upstream to avoid ``versions.patch``, we'll see if they are interested.